tests/conftest.py

```diff
@@ -126,3 +150,179 @@ async def ccn_api_client(
+        sender="0x9319Ad3B7A8E0eE24f2E639c40D8eD124C5520Ba",
+        signature=None,
+        item_type=ItemType.inline,
+        item_content=json.dumps(content),
+        time=timestamp_to_datetime(1619017773.8950577),
+        channel=None,
+        reception_time=timestamp_to_datetime(1619017774),
+        fetched=True,
+        check_message=False,
+        retries=0,
+        next_attempt=dt.datetime(2023, 1, 1),
+    )
+    with session_factory() as session:
+        session.add(pending_message)
+        session.add(
+            MessageStatusDb(
+                item_hash=pending_message.item_hash,
+                status=MessageStatus.PENDING,
+                reception_time=pending_message.reception_time,
+            )
+        )
+        session.commit()
+
+    return pending_message
+
+
+class Volume(Protocol):
+    ref: str
+    use_latest: bool
+
+
+def get_volume_refs(content: ExecutableContent) -> List[Volume]:
+    volumes = []
+
+    for volume in content.volumes:
+        if isinstance(volume, ImmutableVolume):
+            volumes.append(volume)
+
+    if isinstance(content, ProgramContent):
+        volumes += [content.code, content.runtime]
+        if content.data:
+            volumes.append(content.data)
+
+    elif isinstance(content, InstanceContent):
+        if parent := content.rootfs.parent:
+            volumes.append(parent)
+
+    return volumes
+
+
+def insert_volume_refs(session: DbSession, message: PendingMessageDb):
+    """
+    Insert volume references in the DB to make the program processable.
+    """
+
+    content = InstanceContent.parse_raw(message.item_content)
+    volumes = get_volume_refs(content)
+
+    created = pytz.utc.localize(dt.datetime(2023, 1, 1))
+
+    for volume in volumes:
+        # Note: we use the reversed ref to generate the file hash for style points,
+        # but it could be set to any valid hash.
+        file_hash = volume.ref[::-1]
+
+        session.add(StoredFileDb(hash=file_hash, size=1024 * 1024, type=FileType.FILE))
+        session.flush()
+        insert_message_file_pin(
+            session=session,
+            file_hash=volume.ref[::-1],
+            owner=content.address,
+            item_hash=volume.ref,
+            ref=None,
+            created=created,
+        )
+        upsert_file_tag(
+            session=session,
+            tag=FileTag(volume.ref),
+            owner=content.address,
+            file_hash=volume.ref[::-1],
+            last_updated=created,
+        )
+
+
+@pytest.fixture
+def user_balance(
+    session_factory: DbSessionFactory, fixture_instance_message
+) -> AlephBalanceDb:
+    balance = AlephBalanceDb(
+        address="0x9319Ad3B7A8E0eE24f2E639c40D8eD124C5520Ba",
+        chain=Chain.ETH,
+        balance=Decimal(22_192),
+        eth_height=0,
+    )
+
+    with session_factory() as session:
+        session.add(balance)
+        session.commit()
+        insert_volume_refs(session, fixture_instance_message)
```

Comment:
I don't understand why you insert the volumes in this fixture, as it has nothing to do with the balance. I expect 3 fixtures:

* `instance_message`
* `instance_message_with_volumes_in_db` (depends on the first one)
* `user_balance` (depends on nothing).